### PR TITLE
Add SSH wrapper for macOS

### PR DESCRIPTION
**Pull request by sergiou87**
This PR adds the SSH wrapper needed on macOS for Desktop to intercept prompts from SSH via the askpass trampoline.

Since it's only required on macOS, I've tried to disable Windows builds… but that can't be easily done via the `binding.gyp` file. Instead I would need to do _something_ to exclude the new `ssh-wrapper` target before running `node-gyp rebuild`. For now, I just made a dummy exe file on Windows that does nothing.

Something similar happened with the tests, since don't want them to run on Windows. I didn't see a way of excluding certain test suites in certain platforms, so I had to keep one test case that checks `ssh-wrapper.exe` exists and ignore the rest 😢  **EDIT:** and after having some issues with the ssh-wrapper tests on Linux, I disabled them there 😅 

I'm not proud of this, and definitely open to suggestions!!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,7 +48,9 @@ jobs:
         name: Install Windows arm64 node.lib
 
       - name: Install and build
-        run: yarn
```
Comment (sergiou87) [inline .github/workflows/ci.yml:51]: This was installing prebuilt binaries of the trampoline and wrapper 🤯

File: test/ssh-wrapper.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+describe('ssh-wrapper', () => {
+  it('exists and is a regular file', async () =>
+    expect((await stat(sshWrapperPath)).isFile()).toBe(true))
+
+  // On Windows, the binary generated is just useless, so no point to test it.
+  // Also, this won't be used on Linux (for now at least), so don't bother to
+  // run the tests there.
+  if (process.platform !== 'darwin') {
+    return
+  }
```
Comment (sergiou87) [inline test/ssh-wrapper.test.js:19]: Not proud of this 😆

File: src/ssh-wrapper.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#ifdef WINDOWS
+
+int main(int argc, char **argv) {
+  // Not needed on Windows, this will just create a dummy executable
+  return -1;
+}
```
Comment (sergiou87) [inline src/ssh-wrapper.c:6]: Not proud of this 😆

**Review by tidy-dev — APPROVED** [review]
✨  Makes sense to me.. and I don't have any suggestions for another approach for being os specific. :/

**Pull request merged**